src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,30 @@
 package mate.academy;
 
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final Set<EventListener> listeners = ConcurrentHashMap.newKeySet();
+    private final ExecutorService executor = Executors.newCachedThreadPool();
 
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : listeners) {
+            CompletableFuture.runAsync(() -> listener.onEvent(event), executor);
+        }
     }
 
     public void shutdown() {
-
+        executor.shutdown();
     }
```

Comment:
It's good practice to also await termination after shutting down the executor to ensure that all tasks have completed. This is typically done using executor.awaitTermination with a suitable timeout.